bson/bsoncodec/bsoncodec.go

```diff
@@ -123,6 +123,21 @@ type DecodeContext struct {
 	// Ancestor is a bson.M, BSON embedded document values being decoded into an empty interface
 	// will be decoded into a bson.M.
 	Ancestor reflect.Type
+
+	// DocumentDecodeType will decode embedded documents into the defined type, rather than into primitive.D.  This is
+	// work-around for custom typing and is All-Or-None.
```

Comment:
> and is All-Or-None.

What do you mean by this?

Reply:
This is not an accurate descriptor.  I will definitely remove it!